File: mvvm/src/main/java/com/hhh/mvvm/recycler/DiffCallbackProxy.java
package com.hhh.mvvm.recycler;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.DiffUtil;

import com.hhh.mvvm.listener.DiffCallback;

public class DiffCallbackProxy<MODEL extends DiffCallback<MODEL>> extends DiffUtil.ItemCallback<MODEL> {

  @Override
  public boolean areItemsTheSame(@NonNull MODEL oldItem, @NonNull MODEL newItem) {
    return oldItem.areItemsTheSame(newItem);
  }

  @Override
  public boolean areContentsTheSame(@NonNull MODEL oldItem, @NonNull MODEL newItem) {
    return oldItem.areContentsTheSame(newItem);
  }
}

File: core/src/main/java/com/nb/core/DiffCallbackProxy.kt
package com.nb.core

import androidx.recyclerview.widget.DiffUtil

class DiffCallbackProxy<MODEL : DiffCallback<MODEL>> : DiffUtil.ItemCallback<MODEL>() {

    override fun areItemsTheSame(oldItem: MODEL, newItem: MODEL) = oldItem.areItemsTheSame(newItem)

    override fun areContentsTheSame(oldItem: MODEL, newItem: MODEL) =
        oldItem.areContentsTheSame(newItem)
}